state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -1133,6 +1151,8 @@ pub mod pallet {
 				return Err(Error::<T>::NoPrivateChannelExistsForBroker.into())
 			};
 
+			T::Bonder::unbond(&broker_id);
```

Comment:
I think it's worth taking a deeper think what it means to do this here. So far I don't see there should be an edge case since this extrinsic is only callable by a broker. Still, we should care about any edge cases here in which an account could benefit from it by changing account roles between broker and validator.

Reply:
It shouldn't be possible to change account roles if the account is still bonded.

Reply:
Currently this is ensured because the only way to be bonded is to open a private channel, and the only way to unbond is by closing it. And we check for closure in `deregister_as_broker`.